src/main/java/Bob/Parser.java

```diff
@@ -0,0 +1,137 @@
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Locale;
+
+import bob.task.Deadline;
+import bob.task.Event;
+import bob.task.Task;
+import bob.task.Todo;
+
+/**
+ * Represents a parser that parses user input and executes the corresponding
+ * commands.
+ */
+class Parser {
+
+    /**
+     * Parses a string into a LocalDateTime object.
+     *
+     * @param input
+     * @return LocalDateTime
+     * @throws BobException
+     */
+    public static LocalDateTime parseDateTime(String input) throws BobException {
+        try {
+            // Correct format: "dd/MM/yyyy HHmm" (e.g., "02/12/2019 1800")
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm", Locale.ENGLISH);
+            return LocalDateTime.parse(input, formatter);
+        } catch (DateTimeParseException e) {
+            throw new BobException("The date and time format is incorrect: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Command to list all tasks.
+     *
+     * @param tasks
+     */
+    public static String executeListCommand(List<Task> tasks) {
+        String out = "Here are the tasks in your list: \n";
+        for (int i = 0; i < tasks.size(); i++) {
+            out += (i + 1) + ". " + tasks.get(i) + "\n";
+        }
+        return out;
+    }
+
+    /**
+     * Command to mark, unmark, or delete a task.
+     *
+     * @param userInput
+     * @param tasks
+     * @throws BobException
+     * @throws IOException
+     */
+    public static String processTaskModificationCommands(String userInput, List<Task> tasks) 
+            throws BobException, IOException {
+        String[] words = userInput.split(" ");
+        int index = Integer.parseInt(words[1]) - 1;
+
+        if (index < 0 || index >= tasks.size()) {
+            throw new BobException("The task index you provided is out of range.");
+        }
+
+        if (userInput.startsWith("mark")) {
+            tasks.get(index).markAsDone();
+            return "Nice! I've marked this task as done: \n" + tasks.get(index);
+        } else if (userInput.startsWith("unmark")) {
+            tasks.get(index).unmarkAsDone();
+            return "OK, I've marked this task as not done yet: \n" + tasks.get(index);
+        } else if (userInput.startsWith("delete")) {
+            Task removed = tasks.remove(index);
+            return "Noted. I've removed this task: \n" + removed + "\nNow you have " 
+                    + tasks.size() + " tasks in the list.";
+        }
+        Storage.saveTasks(tasks);
+
+        return "Invalid command.";
+    }
+
+    /**
+     * Creates a new task based on the user input and adds it to the list of
+     * tasks.
+     *
+     * @param userInput
+     * @param tasks
+     * @throws BobException
+     * @throws IOException
+     */
+    public static String processTaskCreationCommands(String userInput, List<Task> tasks) 
+            throws BobException, IOException {
```

Comment:
Perhaps you may want to hide the logic of how a task is stored to within this method?
Currently, if this method throws `IOException`, all other usages of this method will need to know that this method performs some sort of IO. If, in the future, the IO is removed, then all the usages will need to be updated to not have try catches.

Instead, it may be better to catch the `IOException` and rethrow as a `BobException`